tests/system/test_enrollment/test_enrollment.py

```diff
@@ -0,0 +1,255 @@
+        elif 'ipv6' in configuration['ip_type']:
+            messages_with_ip = messages_with_ip.replace('AGENT_IP', f"{network['agent_network'][1]}")
+        elif 'dns' in configuration['ip_type']:
+            if 'yes' in configuration['ipv6_enabled']:
+                if 'ipv4' in configuration['agent_network']:
+                    messages_with_ip = messages_with_ip.replace('AGENT_IP', f"{network['agent_network'][0]}")
+                else:
+                    messages_with_ip = messages_with_ip.replace('AGENT_IP', f"{network['agent_network'][1]}")
+            else:
+                messages_with_ip = messages_with_ip.replace('AGENT_IP', f"{network['agent_network'][0]}")
+
+    with open(manager_conf_file, 'w') as file:
+        file.write(new_manager_configuration)
+
+    host_manager.apply_config(manager_conf_file)
+
+    with open(messages_path, 'w') as file:
+        file.write(messages_with_ip)
+
+    yield
+
+    with open(messages_path, 'w') as file:
+        file.write(messages)
+
+    with open(manager_conf_file, 'w') as file:
+        file.write(old_manager_configuration)
+
+
+@pytest.mark.parametrize('test_case', [cases for cases in test_cases_yaml], ids=[cases['name']
+                         for cases in test_cases_yaml])
+def test_agent_enrollment(test_case, get_ip_directions, configure_network, modify_ip_address_conf, clean_environment):
+    '''
+    description: Check if enrollment messages are sent in the correct format
+                 and the agent is registered and connected.
+    wazuh_min_version: 4.4.0
+    parameters:
+        - test_case:
+            type: list
+            brief: List of tests to be performed.
+        - get_ip_directions:
+            type: fixture
+            brief: Get IP from the manager and the agent.
+        - configure_network:
+            type: fixture
+            brief: Configure a custom network environment for testing.
+        - modify_ip_address_conf:
+            type: fixture
+            brief: Add IP to test configuration.
+        - clean_environment:
+            type: fixture
+            brief: Clean environment after every test execution.
+    assertions:
+        - Verify that expected logs are received after registering an agent.
+        - Verify that 'client.keys' are equal in manager and agent.
+        - Verify that the agent is 'Active'
+    input_description: Different use cases are found in the test module and include parameters
+                       for 'agent-auth' messages.
+    expected_output:
+        - '.*Received request for a new agent .* from: AGENT_IP'
+        - '.*Agent key generated for.*'
+        - '.*Server IP Address: MANAGER_IP'
+        - '.*Requesting a key from server: MANAGER_IP'
+        - '.*Registering agent to unverified manager'
+        - '.*Using agent name as:*'
+        - '.*Waiting for server reply'
+        - '.*Valid key received'
+        - '.*Waiting .* seconds before server connection'
+    tags:
+        - authd
```

Comment:
In agent part there are more modules/daemons implicated.

Reply:
Done in 8243963